DataBase.h

```diff
@@ -51,6 +51,11 @@
 #include <mysql.h>
 #include <update_starter.h>
 
+#ifndef LIBMARIADB
+#ifndef my_bool
+typedef bool my_bool;
+#endif
+#endif
```

Comment:
Have you tested this on older MySQL versions? Because (at least in 5.7) they do not have a macro for that, just a typedef:
https://github.com/mysql/mysql-server/blob/mysql-5.7.31/include/mysql.h#L49
And their typedef is pointing to `char`, which conflicts with what you added here (and this is a compilation error).

People on the Internet are checking the version, like here:
https://github.com/zussel/matador/commit/9869f4ede37be748102fa92fce55a84b016902f0#diff-c4fe758407c037ec2b8fe051215fb3cbR8-R10

Another option is to stop using my_bool in our code. There is just one single occurrence in DataBaseUtils.cpp. We just need to detect somehow what boolean type mysql is using, for instance with something like:
```diff
-               my_bool my_auto_reconnect=1;
+               decltype(mysql_thread_init()) my_auto_reconnect=1;
                if (mysql_options(conn_pool[loop].db,MYSQL_OPT_RECONNECT,&my_auto_reconnect) !=0)
```
If you want to keep it C++03 compatible for any reasons, this will be a bit more complicated (may require overloading, tag dispatching or SFINAE).

Reply:
@mliszcz I only tested with debian buster. I'll do a test with debian wheezy.

Reply:
@t-b did you use packages from debian repos? I don't know what wheezy has but it looks like neither buster nor stretch provide MySQL. libmysqlclient-dev is alias for mariadb package: https://packages.debian.org/buster/default-libmysqlclient-dev
And if you check who provides `mysql.h`, there is only libmariadb-dev: https://packages.debian.org/search?searchon=contents&keywords=mysql.h&mode=path&suite=stable&arch=any

Reply:
@mliszcz Thanks for the review. Indeed this breaks with older mysql versions, as I just tested with debian wheezy. I've now revised the code and only define my_bool for newer MYSQL versions.